### PR TITLE
Add comprehensive unit tests

### PR DESCRIPTION
**Pull request by Gudahtt**
Unit tests have been added to test decrypting data that was encrypted using an older version of this library. Previously encryption and decryption were only tested together, so if a breaking change was made to both encryption and decryption, it may have been missed. Now we should have some guarantee of stability.

Tests have been added for the other untested exported functions as well, which are 'keyFromPassword', 'encryptWithKey', 'decryptWithKey', and 'generateSalt'.

The ESLint configuration was updated to state that these tests are expected to run in a browser, because the `atob` API was required for testing `generateSalt`.

### CODE REVIEW

**Comment by Gudahtt** [comment]
These tests were written against the latest published version of this package, v2.0.3. You can check them out on the branch `add-addtional-tests`, here: https://github.com/MetaMask/browser-passworder/blob/add-additional-tests/test/index.js

That branch demonstrates that these tests correctly represent the behavior in production right now.

File: test/index.js [hunk 1]
```diff
@@ -62,3 +126,217 @@ QUnit.test(
     }
   },
 );
+
+const encryptedData = {
```
Comment (brad-decker) [inline test/index.js:130]: [Q] This is the data that was encrypted using an older version of the library? might be good to just add a comment around the importance of not modifying this structure and what its intended purpose is.

**Review by brad-decker — CHANGES_REQUESTED** [review]
tests look good to me, just a question/ possible documentation request to provide a warning to those who would seek to modify the encryptedData object

File: test/index.js (hunk 1 above)
Comment (Gudahtt) [reply test/index.js:130]: Good idea!

**Review by brad-decker — APPROVED** [review]

**Pull request merged**